module/modmanager/manager.go

```diff
@@ -87,12 +94,19 @@ type addedResource struct {
 
 // Manager is the root structure for the module system.
 type Manager struct {
-	mu                      sync.RWMutex
-	logger                  logging.Logger
-	modules                 map[string]*module
-	parentAddr              string
-	rMap                    map[resource.Name]*module
-	untrustedEnv            bool
+	mu           sync.RWMutex
+	logger       logging.Logger
+	modules      map[string]*module
+	parentAddr   string
+	rMap         map[resource.Name]*module
+	untrustedEnv bool
+	// viamHomeDir is the absolute path to the viam home directory. Ex: /home/walle/.viam
+	// it is empty if the modmanageroptions.Options.viamHomeDir was empty
+	viamHomeDir string
```

Comment:
Because the home dir is required for env var insertion. (and I would prefer to store the moduleDataParentDir instead of storing the robot id and re-evaluating the parentDir every time it is needed... that is a personal preference though and I am open to your stylistic opinions.)